ocs_ci/deployment/flexy.py

```diff
@@ -379,25 +369,31 @@ def deploy(self, log_level=""):
         if log_level:
             pass
         cmd = self.build_install_cmd()
-        flexy_err = None
         # Ensure that flexy workdir will be copied to cluster dir even when
         # Flexy itself fails.
         try:
             self.run_container(cmd)
         except Exception as err:
             logger.error(err)
-            flexy_err = err
+            raise
+        finally:
+            self.flexy_backup_work_dir()
+
         self.flexy_post_processing()
-        if flexy_err:
-            raise flexy_err
 
     def destroy(self):
         """
         Invokes flexy container with 'destroy' argument
 
         """
         cmd = self.build_destroy_cmd()
-        self.run_container(cmd)
+        try:
+            self.run_container(cmd)
+        except Exception as err:
+            logger.error(err)
+            raise
+        finally:
+            self.flexy_backup_work_dir()
```

Comment:
I might be missing something but why do we need to backup work dir during after destroy ?

Reply:
My intention is, to have the _cluster dir_ consistent with the state of the cluster. (So if the cluster is destroyed, the _cluster dir_ should reflect it I think.)